app/jenkins-for-jira-ui/src/components/ServerManagement/ServerManagement.tsx

```diff
@@ -1,34 +1,120 @@
-import React from 'react';
+import React, { useState, useRef, useEffect } from 'react';
 import PageHeader from '@atlaskit/page-header';
 import { ButtonGroup } from '@atlaskit/button';
 import Button from '@atlaskit/button/standard-button';
-import { headerContainer } from '../JenkinsServerList/JenkinsServerList.styles';
-import { mainPageContainer } from './ServerManagement.styles';
+import TextArea from '@atlaskit/textarea';
+import { cx } from '@emotion/css';
+import { serverManagementContainer, modalHeaderContainer, shareModalInstruction } from './ServerManagement.styles';
 import { ConnectionPanel } from '../ConnectionPanel/ConnectionPanel';
 import { TopPanel } from './TopPanel/TopPanel';
+import { JenkinsModal } from '../JenkinsServerList/ConnectedServer/JenkinsModal';
+import { fetchGlobalPageUrl } from '../../api/fetchGlobalPageUrl';
+
+export const getSiteNameFromUrl = (url: string): string => {
+	try {
+		const urlObject = new URL(url);
+		return urlObject.hostname;
+	} catch (error) {
+		console.error('Error extracting site name:', error);
+		return '';
+	}
+};
 
 const ServerManagement = (): JSX.Element => {
+	const [showSharePage, setshowSharePage] = useState<boolean>(false);
+	const [isCopiedToClipboard, setIsCopiedToClipboard] = useState(false);
+	const [globalPageUrl, setGlobalPageUrl] = useState<string>('');
+	const textAreaRef = useRef<HTMLTextAreaElement>(null);
+
+	const handleShowSharePageModal = async () => {
+		setshowSharePage(true);
+	};
+
+	const handleCloseShowSharePageModal = async () => {
+		setshowSharePage(false);
+	};
+
+	const handleCopyToClipboard = async () => {
+		if (textAreaRef.current) {
+			textAreaRef.current.select();
+			document.execCommand('copy');
+			textAreaRef.current.setSelectionRange(textAreaRef.current.value.length, textAreaRef.current.value.length);
+
+			setIsCopiedToClipboard(true);
+
+			setTimeout(() => {
+				setIsCopiedToClipboard(false);
+			}, 2000);
+		}
+	};
+
+	useEffect(() => {
+		const fetchData = async () => {
+			try {
+				const url = await fetchGlobalPageUrl();
+				setGlobalPageUrl(url);
+			} catch (error) {
+				console.error('Error fetching data:', error);
+			}
+		};
+
+		fetchData();
+	}, []);
+
 	const pageHeaderActions = (
 		<ButtonGroup>
 			{/* TODO handle empty state - ARC-2730 connection wizard */}
-			<Button appearance="primary">
-				Connect a new Jenkins server
-			</Button>
-			{/* TODO - ARC-2723 share modal */}
-			<Button>Share page</Button>
+			<Button appearance="primary">Connect a new Jenkins server</Button>
+			<Button onClick={() => handleShowSharePageModal()}>Share page</Button>
 		</ButtonGroup>
 	);
 
-	// TODO handle empty state - ARC-2730 connection wizard
+	const sharePageMessage =
+		`Hi there,
+
+Jenkins for Jira is now installed and connected on ${getSiteNameFromUrl(globalPageUrl)}.
```

Comment:
Alignment for this looks whack but, unfortunately, I don't think there's a way around it. Indenting means indenting the content in the text area and everything needs to be left-aligned.